src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/AWSSecretsManagerPlugin.java

```diff
@@ -179,7 +179,19 @@ public void openInitialConnection(ConnectionUrl connectionUrl) throws SQLExcepti
    */
   private boolean isLoginUnsuccessful(SQLException exception) {
     this.logger.logTrace("Login failed. SQLState=" + exception.getSQLState(), exception);
-    return SQLSTATE_ACCESS_ERROR.equals(exception.getSQLState());
+
+    Throwable throwable = exception;
+    while (throwable != null) {
+      if (throwable instanceof SQLException) {
```

Comment:
The internal exception which is wrapped in java.sql.SQLNonTransientConnectionException is of Type :- software.aws.rds.jdbc.mysql.shading.com.mysql.cj.exceptions.CJException.
This exception extends RuntimeException and not SQLException, therefore, the internal exception will be skipped and secrets won't get rotated.

The custom codefix which I tested :- 
`private boolean isLoginUnsuccessful(SQLException exception) {
    this.logger.logTrace("Login failed. SQLState=" + exception.getSQLState(), exception);
    Throwable t = exception;
    do {
      if (t instanceof SQLException) {
        if (((SQLException) t).getSQLState().equals(SQLSTATE_ACCESS_ERROR)) {
          return true;
        }
      }
      if (t instanceof CJException) {
        if (((CJException) t).getSQLState().equals(SQLSTATE_ACCESS_ERROR)) {
          return true;
        }
      }
        t = t.getCause();
    } while (t != null);
    return false;
  }`


-- Anmolpreet Singh (Chegg Inc)